core/src/sumeragi/main_loop.rs

```diff
@@ -797,10 +806,11 @@ pub(crate) fn run<F: FaultInjection>(
             &mut voting_block,
             &mut voting_signatures,
             &mut round_start_time,
+            &mut last_view_change_time,
             &mut view_change_time,
         );
 
-        if round_start_time.elapsed() > view_change_time {
+        if node_wants_block && last_view_change_time.elapsed() > view_change_time {
```

Comment:
If you've seen my last comment, I had a small change of heart since then. It's good that you introduced `last_view_change_time` so that block is created immediately after a view change, i.e. the round is not restarted. However, I am still advocating that we should do view changes when network is idle as per [these](https://github.com/hyperledger/iroha/pull/3097/files#r1091892835) arguments. This is the only change I advocate in this PR:

```suggestion
        if last_view_change_time.elapsed() > view_change_time {
```

Reply:
alright then.

Reply:
you make the suggested change, I'll open an issue so we can discuss